File: create_service_cards.py
#!/usr/bin/env python3
"""
创建 MARKET LINK EXPRESS 服务卡片
基于用户提供的图片创建服务卡片版本
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_service_cards():
    """创建服务卡片"""
    
    try:
        # 1. 创建服务卡片背景
        card_width = 400
        card_height = 300
        
        # 使用渐变背景
        background = Image.new('RGBA', (card_width, card_height), (176, 211, 232, 255))
        
        # 2. 加载公司 LOGO
        if os.path.exists('public/logo.png'):
            company_logo = Image.open('public/logo.png')
            logo_size = (60, 60)
            company_logo_resized = company_logo.resize(logo_size, Image.Resampling.LANCZOS)
            
            # 将 LOGO 放在左上角
            background.paste(company_logo_resized, (20, 20), company_logo_resized)
        
        # 3. 添加标题
        draw = ImageDraw.Draw(background)
        
        # 尝试加载字体
        try:
            title_font = ImageFont.truetype("arial.ttf", 24)
            subtitle_font = ImageFont.truetype("arial.ttf", 16)
        except:
            title_font = ImageFont.load_default()
            subtitle_font = ImageFont.load_default()
        
        # 标题
        title_text = "MARKET LINK EXPRESS"
        title_bbox = draw.textbbox((0, 0), title_text, font=title_font)
        title_width = title_bbox[2] - title_bbox[0]
        title_x = (card_width - title_width) // 2
        draw.text((title_x, 100), title_text, fill=(255, 255, 255, 255), font=title_font)
        
        # 副标题
        subtitle_text = "曼德勒最萌快递服务"
        subtitle_bbox = draw.textbbox((0, 0), subtitle_text, font=subtitle_font)
        subtitle_width = subtitle_bbox[2] - subtitle_bbox[0]
        subtitle_x = (card_width - subtitle_width) // 2
        draw.text((subtitle_x, 130), subtitle_text, fill=(255, 255, 255, 255), font=subtitle_font)
        
        # 4. 添加服务特色
        features = [
            "⚡ 实时小猫追踪",
            "🛡️ 快速安全送达", 
            "📱 可爱客服体验"
        ]
        
        feature_y = 170
        for feature in features:
            draw.text((50, feature_y), feature, fill=(255, 255, 255, 255), font=subtitle_font)
            feature_y += 25
        
        # 5. 添加二维码区域
        qr_area_size = 80
        qr_x = card_width - qr_area_size - 20
        qr_y = card_height - qr_area_size - 20
        
        # 绘制二维码背景
        qr_bg = Image.new('RGBA', (qr_area_size, qr_area_size), (255, 255, 255, 255))
        background.paste(qr_bg, (qr_x, qr_y))
        
        # 添加二维码文字
        qr_text = "扫码下载"
        qr_text_bbox = draw.textbbox((0, 0), qr_text, font=subtitle_font)
        qr_text_width = qr_text_bbox[2] - qr_text_bbox[0]
        qr_text_x = qr_x + (qr_area_size - qr_text_width) // 2
        draw.text((qr_text_x, qr_y + qr_area_size + 5), qr_text, fill=(255, 255, 255, 255), font=subtitle_font)
        
        # 6. 保存服务卡片
        background.save('service_cards.png', 'PNG')
        
        print("✅ 服务卡片创建完成!")
        print("📁 生成的文件: service_cards.png")
        
        return True
        
    except Exception as e:
        print(f"❌ 创建服务卡片时出错: {str(e)}")
        return False

def create_simple_service_card():
    """创建简单的服务卡片"""
    
    try:
        # 创建服务卡片
        card_width = 350
        card_height = 250
        
        # 使用渐变背景
        background = Image.new('RGBA', (card_width, card_height), (176, 211, 232, 255))
        
        # 添加渐变效果
        draw = ImageDraw.Draw(background)
        for y in range(card_height):
            alpha = int(255 * (1 - y / card_height * 0.3))
            color = (176, 211, 232, alpha)
            draw.line([(0, y), (card_width, y)], fill=color)
        
        # 加载公司 LOGO
        if os.path.exists('public/logo.png'):
            company_logo = Image.open('public/logo.png')
            logo_size = (50, 50)
            company_logo_resized = company_logo.resize(logo_size, Image.Resampling.LANCZOS)
            
            # 将 LOGO 放在左上角
            background.paste(company_logo_resized, (15, 15), company_logo_resized)
        
        # 添加标题
        try:
            title_font = ImageFont.truetype("arial.ttf", 20)
            subtitle_font = ImageFont.truetype("arial.ttf", 14)
        except:
            title_font = ImageFont.load_default()
            subtitle_font = ImageFont.load_default()
        
        # 标题
        title_text = "MARKET LINK EXPRESS"
        draw.text((80, 20), title_text, fill=(255, 255, 255, 255), font=title_font)
        
        # 副标题
        subtitle_text = "曼德勒最萌快递服务"
        draw.text((80, 45), subtitle_text, fill=(255, 255, 255, 255), font=subtitle_font)
        
        # 添加服务特色
        features = [
            "⚡ 实时小猫追踪",
            "🛡️ 快速安全送达", 
            "📱 可爱客服体验"
        ]
        
        feature_y = 80
        for feature in features:
            draw.text((20, feature_y), feature, fill=(255, 255, 255, 255), font=subtitle_font)
            feature_y += 25
        
        # 添加二维码区域
        qr_area_size = 60
        qr_x = card_width - qr_area_size - 15
        qr_y = card_height - qr_area_size - 15
        
        # 绘制二维码背景
        qr_bg = Image.new('RGBA', (qr_area_size, qr_area_size), (255, 255, 255, 255))
        background.paste(qr_bg, (qr_x, qr_y))
        
        # 添加二维码文字
        qr_text = "扫码下载"
        draw.text((qr_x, qr_y + qr_area_size + 5), qr_text, fill=(255, 255, 255, 255), font=subtitle_font)
        
        # 保存服务卡片
        background.save('simple_service_card.png', 'PNG')
        
        print("✅ 简单服务卡片创建完成!")
        print("📁 生成的文件: simple_service_card.png")
        
        return True
        
    except Exception as e:
        print(f"❌ 创建简单服务卡片时出错: {str(e)}")
        return False

if __name__ == "__main__":
    print("🚀 开始创建 MARKET LINK EXPRESS 服务卡片...")
    
    # 创建两种版本的服务卡片
    success1 = create_service_cards()
    success2 = create_simple_service_card()
    
    if success1 and success2:
        print("\n🎉 服务卡片创建完成!")
        print("📋 生成的文件:")
        print("   - service_cards.png (完整版本)")
        print("   - simple_service_card.png (简化版本)")
        print("\n📋 下一步:")
        print("1. 检查生成的图片")
        print("2. 将图片复制到应用资源目录")
        print("3. 更新应用中的服务卡片")
    else:
        print("\n❌ 创建失败，请检查错误信息")

File: create_service_cards.js
#!/usr/bin/env node
/**
 * 创建 MARKET LINK EXPRESS 服务卡片
 * 使用 Node.js 和 Canvas API 生成服务卡片
 */

const fs = require('fs');
const path = require('path');

// 检查是否安装了 canvas
let canvas, Image;
try {
    canvas = require('canvas');
    Image = canvas.Image;
} catch (error) {
    console.log('❌ 需要安装 canvas 包: npm install canvas');
    console.log('📋 或者使用其他方法处理图片');
    process.exit(1);
}

function createServiceCard() {
    try {
        // 创建画布
        const width = 400;
        const height = 300;
        const canvasElement = canvas.createCanvas(width, height);
        const ctx = canvasElement.getContext('2d');
        
        // 创建渐变背景
        const gradient = ctx.createLinearGradient(0, 0, 0, height);
        gradient.addColorStop(0, '#b0d3e8');
        gradient.addColorStop(1, '#558cea');
        
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 0, width, height);
        
        // 添加公司 LOGO (如果有的话)
        if (fs.existsSync('public/logo.png')) {
            try {
                const logoImg = new Image();
                logoImg.src = fs.readFileSync('public/logo.png');
                
                // 调整 LOGO 大小
                const logoSize = 50;
                ctx.drawImage(logoImg, 15, 15, logoSize, logoSize);
            } catch (error) {
                console.log('⚠️ 无法加载 LOGO 图片');
            }
        }
        
        // 添加标题
        ctx.fillStyle = 'white';
        ctx.font = 'bold 20px Arial';
        ctx.textAlign = 'left';
        ctx.fillText('MARKET LINK EXPRESS', 80, 35);
        
        // 添加副标题
        ctx.font = '14px Arial';
        ctx.fillText('曼德勒最萌快递服务', 80, 55);
        
        // 添加服务特色
        const features = [
            '⚡ 实时小猫追踪',
            '🛡️ 快速安全送达',
            '📱 可爱客服体验'
        ];
        
        let y = 90;
        features.forEach(feature => {
            ctx.fillText(feature, 20, y);
            y += 25;
        });
        
        // 添加二维码区域
        const qrSize = 60;
        const qrX = width - qrSize - 15;
        const qrY = height - qrSize - 15;
        
        // 绘制二维码背景
        ctx.fillStyle = 'white';
        ctx.fillRect(qrX, qrY, qrSize, qrSize);
        
        // 添加二维码边框
        ctx.strokeStyle = '#ccc';
        ctx.lineWidth = 1;
        ctx.strokeRect(qrX, qrY, qrSize, qrSize);
        
        // 添加二维码文字
        ctx.fillStyle = 'white';
        ctx.font = '12px Arial';
        ctx.textAlign = 'center';
        ctx.fillText('扫码下载', qrX + qrSize/2, qrY + qrSize + 20);
        
        // 保存图片
        const buffer = canvasElement.toBuffer('image/png');
        fs.writeFileSync('service_cards.png', buffer);
        
        console.log('✅ 服务卡片创建完成!');
        console.log('📁 生成的文件: service_cards.png');
        
        return true;
        
    } catch (error) {
        console.log(`❌ 创建服务卡片时出错: ${error.message}`);
        return false;
    }
}

function createSimpleServiceCard() {
    try {
        // 创建较小的服务卡片
        const width = 300;
        const height = 200;
        const canvasElement = canvas.createCanvas(width, height);
        const ctx = canvasElement.getContext('2d');
        
        // 创建渐变背景
        const gradient = ctx.createLinearGradient(0, 0, 0, height);
        gradient.addColorStop(0, '#b0d3e8');
        gradient.addColorStop(1, '#558cea');
        
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 0, width, height);
        
        // 添加公司 LOGO
        if (fs.existsSync('public/logo.png')) {
            try {
                const logoImg = new Image();
                logoImg.src = fs.readFileSync('public/logo.png');
                
                const logoSize = 40;
                ctx.drawImage(logoImg, 10, 10, logoSize, logoSize);
            } catch (error) {
                console.log('⚠️ 无法加载 LOGO 图片');
            }
        }
        
        // 添加标题
        ctx.fillStyle = 'white';
        ctx.font = 'bold 16px Arial';
        ctx.textAlign = 'left';
        ctx.fillText('MARKET LINK EXPRESS', 60, 30);
        
        // 添加副标题
        ctx.font = '12px Arial';
        ctx.fillText('曼德勒最萌快递服务', 60, 45);
        
        // 添加服务特色
        const features = [
            '⚡ 实时追踪',
            '🛡️ 安全送达',
            '📱 客服体验'
        ];
        
        let y = 70;
        features.forEach(feature => {
            ctx.fillText(feature, 15, y);
            y += 20;
        });
        
        // 添加二维码区域
        const qrSize = 50;
        const qrX = width - qrSize - 10;
        const qrY = height - qrSize - 10;
        
        // 绘制二维码背景
        ctx.fillStyle = 'white';
        ctx.fillRect(qrX, qrY, qrSize, qrSize);
        
        // 添加二维码边框
        ctx.strokeStyle = '#ccc';
        ctx.lineWidth = 1;
        ctx.strokeRect(qrX, qrY, qrSize, qrSize);
        
        // 添加二维码文字
        ctx.fillStyle = 'white';
        ctx.font = '10px Arial';
        ctx.textAlign = 'center';
        ctx.fillText('扫码下载', qrX + qrSize/2, qrY + qrSize + 15);
        
        // 保存图片
        const buffer = canvasElement.toBuffer('image/png');
        fs.writeFileSync('simple_service_card.png', buffer);
        
        console.log('✅ 简单服务卡片创建完成!');
        console.log('📁 生成的文件: simple_service_card.png');
        
        return true;
        
    } catch (error) {
        console.log(`❌ 创建简单服务卡片时出错: ${error.message}`);
        return false;
    }
}

// 主函数
function main() {
    console.log('🚀 开始创建 MARKET LINK EXPRESS 服务卡片...');
    
    const success1 = createServiceCard();
    const success2 = createSimpleServiceCard();
    
    if (success1 && success2) {
        console.log('\n🎉 服务卡片创建完成!');
        console.log('📋 生成的文件:');
        console.log('   - service_cards.png (完整版本)');
        console.log('   - simple_service_card.png (简化版本)');
        console.log('\n📋 下一步:');
        console.log('1. 检查生成的图片');
        console.log('2. 将图片复制到应用资源目录');
        console.log('3. 更新应用中的服务卡片');
    } else {
        console.log('\n❌ 创建失败，请检查错误信息');
    }
}

// 运行主函数
main();
